### PR TITLE
feat: Milestone 2 — Headless mode, output formatters, code review pipeline

### PR DESCRIPTION
**Pull request by julianshen**
## Summary

- Add `--headless` mode for non-interactive CLI execution with `--prompt`, `--file`, and stdin input
- Add JSON and Markdown output formatters (`--output json|markdown`)
- Add code review pipeline (`--mode=code-review --diff HEAD~1..HEAD`)
- Add execution controls: `--max-turns`, `--timeout`, `--tools` whitelist
- Exit code 0 for success, 1 for errors

### New packages
- `internal/output/` — `RunResult` types, `Formatter` interface, JSON and Markdown formatters
- `internal/runner/` — Input resolver, `HeadlessRunner` with `TurnFunc` abstraction
- `internal/pipeline/` — Git diff extraction, code review prompt builder

### Modified files
- `cmd/rubichan/main.go` — Headless flags, routing, tool whitelist
- `.gitignore` — Fixed pattern to avoid matching `cmd/rubichan/` directory

## Test plan
- [x] All 11 packages pass (`go test ./...`)
- [x] Input resolver: 9 tests covering flag/file/stdin priority, edge cases
- [x] Headless runner: 4 tests with mock TurnFunc (basic, tool calls, error, timeout)
- [x] Output formatters: 7 tests for JSON and Markdown output
- [x] Pipeline: 5 tests with temp git repos for diff extraction and prompt building
- [x] CLI helpers: 7 tests for tool whitelist parsing
- [x] Smoke test: JSON output via `--prompt`
- [x] Smoke test: Markdown output via `--prompt`
- [x] Smoke test: stdin pipe input
- [x] Smoke test: code review mode with `--diff`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Headless/non-interactive mode with CLI flags for prompt/file/stdin, mode, output, diff, timeout, and tool selection.
  * Code-review workflow that extracts diffs and builds review prompts.
  * JSON and Markdown output formatters and a structured run result for headless runs.
  * Tool whitelisting to selectively enable tools in headless mode.

* **Tests**
  * Unit and integration tests covering input resolution, diff extraction, formatters, headless runner, and helper logic.

* **Documentation**
  * Added design and plan docs describing headless flow, formatters, and testing.

* **Chore**
  * Tightened .gitignore rule to apply only at repository root.

### CODE REVIEW

File: internal/pipeline/diff.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+	cmd := exec.CommandContext(ctx, "git", "diff", diffRange)
+	cmd.Dir = dir
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("git diff failed: %s: %w", strings.TrimSpace(string(out)), err)
+	}
+
+	return string(out), nil
```
[the remark this answers is not in the document]
Comment (julianshen) [reply internal/pipeline/diff.go]: Fixed in e87aab7. Switched to `cmd.Output()` (stdout only). Stderr is extracted from `*exec.ExitError` on failure for the error message.
